src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -73,6 +74,13 @@ export class BreadcrumbsComponent implements OnDestroy {
           .subscribe((x) => {
             this.finished = true;
             this.crumbs = x;
+            const curLocation = this.crumbs
+              .map((crumb: IBreadcrumb) => {
+                return crumb.text.startsWith('http') ? '' : crumb.text;
+              })
+              .join(' > ');
+            const title = 'Ceph' + (curLocation.length > 0 ? ': ' + curLocation : '');
+            this.titleService.setTitle(title);
```

Comment:
Following the KISS principle, instead of adding more complexity to the constructor, let's isolate the page title setup logic in a method that also has the benefit of being more testable:
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts
index a346f49aac8..35d06553a37 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts
@@ -74,13 +74,7 @@ export class BreadcrumbsComponent implements OnDestroy {
           .subscribe((x) => {
             this.finished = true;
             this.crumbs = x;
-            const curLocation = this.crumbs
-              .map((crumb: IBreadcrumb) => {
-                return crumb.text.startsWith('http') ? '' : crumb.text;
-              })
-              .join(' > ');
-            const title = 'Ceph' + (curLocation.length > 0 ? ': ' + curLocation : '');
-            this.titleService.setTitle(title);
+            this.titleService.setTitle(this.getTitleFromCrumbs(this.crumbs));
           });
       });
   }
@@ -146,4 +140,13 @@ export class BreadcrumbsComponent implements OnDestroy {
 
     return of(value as T);
   }
+
+  private getTitleFromCrumbs(crumbs: IBreadcrumb[]): string {
+    return `Ceph ${crumbs
+      .map((crumb: IBreadcrumb) => {
+        return crumb.text.startsWith('http') ? '' : crumb.text;
+      })
+      .toString()
+      .replace(/,/g, ' > ')}`;
+  }
 }
```

Reply:
@alfonsomthd As suggested I have separated  page title logic from constructor into a method and implemented some unit test for the same.